### PR TITLE
Port of port of Flora code improvement

### PR DESCRIPTION
**Pull request by Alterist**
🆑 
add: Adds Trees, cutting them down, and burning them
imageadd: Ports tg's jungle trees
/ 🆑 
Ports /tg/ jungle trees
Ports some /tg/ code regarding chopping trees
Ports Xmas trees
General improvement regarding general flora, allowing for welders, flamers and explosions to destroy them
Adds Flamer/Tree interaction, allowing them to be set on fire and be destroyed

### CODE REVIEW

**Comment by Arachnidnexus** [comment]
Finally we can get real jungles.

**Comment by Alterist** [comment]
fixed runtimes andinvisible xmas trees
minor numbers adjustment
adds fire overlays for all trees

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -1,48 +1,222 @@
...
+	. = ..()
+	processing_objects.Add(src)
+	return TRUE
+
+/obj/structure/flora/tree/attackby(obj/item/W, mob/user, params)
+	if(W.sharp && W.force > 0)
+		if(W.hitsound)
+			playsound(get_turf(src), W.hitsound, 100, 0, 0)
+		user.visible_message("<span class='notice'>[user] begins to cut down [src] with [W].</span>","<span class='notice'>You begin to cut down [src] with [W].</span>", "You hear the sound of sawing.")
+		if(!do_after(usr, 100, TRUE, 5, BUSY_ICON_BUILD)) //5 seconds with 20 force, 8 seconds with a hatchet, 20 seconds with a shard.
```
Comment (Rohesie) [inline code/game/objects/structures/flora.dm]: The description is not correct here, is it? I seems to be 10 seconds regardless of force/item.

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -1,48 +1,222 @@
...
+	. = ..()
+	processing_objects.Add(src)
+	return TRUE
+
+/obj/structure/flora/tree/attackby(obj/item/W, mob/user, params)
+	if(W.sharp && W.force > 0)
+		if(W.hitsound)
+			playsound(get_turf(src), W.hitsound, 100, 0, 0)
+		user.visible_message("<span class='notice'>[user] begins to cut down [src] with [W].</span>","<span class='notice'>You begin to cut down [src] with [W].</span>", "You hear the sound of sawing.")
+		if(!do_after(usr, 100, TRUE, 5, BUSY_ICON_BUILD)) //10 seconds to cut down a tree
```
Comment (Rohesie) [inline code/game/objects/structures/flora.dm]:
```suggestion
		if(!do_after(usr, 10 SECONDS, TRUE, 5, BUSY_ICON_BUILD)) //10 seconds to cut down a tree
```

Comment (Rohesie) [reply code/game/objects/structures/flora.dm]: You could also do something like this if you want to preserve the varied time function:
```
var/cut_force = min(1, W.force)
var/cutting_time = CLAMP(5, 20, 100/cut_force) SECONDS
if(!do_after(usr, cutting_time , TRUE, 5, BUSY_ICON_BUILD))
```
In CLAMP you put the first two values for the min and max range. Could also tie it to engie skill. Or you could keep it all as is as well. :p

**Comment by Rohesie** [comment]
Ah, but you'll need to replace every instance of `cdel` for `qdel`, and of `Dispose` for `Destroy`, due to the stonedmc changes.

**Pull request merged**